I keep getting this exception from a monitored EAP6.

```
18:13:59,020 ERROR [org.hawkular.agent.monitor.protocol.EndpointService] (Hawkular-WildFly-Agent-Scheduler-Metrics-3) HAWKMONITOR010049: Could not access resources of endpoint [DMREndpointService[Endpoint[remote-eap6]:[remote://localhost:10999]]]: org.hawkular.agent.monitor.protocol.ProtocolException: Unsuccessful fetching DMR attribute [AverageGetTime]
    at org.hawkular.agent.monitor.protocol.dmr.DMRDriver.fetchAttribute(DMRDriver.java:158)
    at org.hawkular.agent.monitor.protocol.EndpointService.measureMetrics(EndpointService.java:447)
    at org.hawkular.agent.monitor.scheduler.MetricsCollector.run(MetricsCollector.java:60)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
    at org.jboss.threads.JBossThread.run(JBossThread.java:320)
Caused by: org.hawkular.dmr.api.OperationFailureException: Could not perform operation [read-attribute]: JBAS014792: Unknown attribute AverageGetTime
    at org.hawkular.dmr.api.OperationBuilder$OperationResult.assertSuccess(OperationBuilder.java:422)
    at org.hawkular.agent.monitor.protocol.dmr.DMRDriver.fetchAttribute(DMRDriver.java:155)
    ... 8 more
```

Not found on EAP6:

```
[standalone@localhost:10999 /] ls /subsystem=datasources/data-source=ExampleDS/statistics=pool
ActiveCount=0             DestroyedCount=0          MaxWaitTime=0             
AvailableCount=0          InUseCount=0              TimedOut=0                
AverageBlockingTime=0     MaxCreationTime=0         TotalBlockingTime=0       
AverageCreationTime=0     MaxUsedCount=0            TotalCreationTime=0       
CreatedCount=0            MaxWaitCount=0            statistics-enabled=false 
```

Is there a way to silence it without affecting other monitored servers?

--- a/hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java
+++ b/hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java
@@ -935,6 +935,7 @@ public class MonitorServiceConfigurationBuilder {
                             RemoteDMRAttributes.SET_AVAIL_ON_SHUTDOWN);
                     Avail setAvailOnShutdown = (setAvailOnShutdownStr == null) ? null
                             : Avail.valueOf(setAvailOnShutdownStr);
+                    String protocol = getString(remoteDMRValueNode, context, RemoteDMRAttributes.PROTOCOL);
                     String host = getString(remoteDMRValueNode, context, RemoteDMRAttributes.HOST);
                     int port = getInt(remoteDMRValueNode, context, RemoteDMRAttributes.PORT);
                     String username = getString(remoteDMRValueNode, context, RemoteDMRAttributes.USERNAME);
@@ -953,7 +954,9 @@ public class MonitorServiceConfigurationBuilder {
                         log.debugf("Using SSL with no security realm - will rely on the JVM truststore: " + name);
                     }
 
-                    String protocol = useSsl ? "https-remoting" : "http-remoting";
+                    if (protocol == null) {
+                        protocol = useSsl ? "https-remoting" : "http-remoting";
+                    }
                     ConnectionData connectionData = new ConnectionData(protocol, host, port, username, password);
                     EndpointConfiguration endpoint = new EndpointConfiguration(name, enabled, resourceTypeSets,
                             connectionData, securityRealm, setAvailOnShutdown, tenantId, metricIdTemplate, metricTags,

--- a/hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/RemoteDMRAttributes.java
+++ b/hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/RemoteDMRAttributes.java
@@ -35,6 +35,13 @@ public interface RemoteDMRAttributes {
                     .addFlag(AttributeAccess.Flag.RESTART_NONE)
                     .build();
 
+    SimpleAttributeDefinition PROTOCOL = new SimpleAttributeDefinitionBuilder("protocol",
+            ModelType.STRING)
+                    .setAllowNull(true)
+                    .setAllowExpression(true)
+                    .addFlag(AttributeAccess.Flag.RESTART_RESOURCE_SERVICES)
+                    .build();
+
     SimpleAttributeDefinition HOST = new SimpleAttributeDefinitionBuilder("host",
             ModelType.STRING)
                     .setAllowNull(false)
@@ -115,6 +122,7 @@ public interface RemoteDMRAttributes {
 
     AttributeDefinition[] ATTRIBUTES = {
             ENABLED,
+            PROTOCOL,
             HOST,
             PORT,
             USERNAME,

--- a/hawkular-wildfly-agent/src/main/resources/org/hawkular/agent/monitor/extension/LocalDescriptions.properties
+++ b/hawkular-wildfly-agent/src/main/resources/org/hawkular/agent/monitor/extension/LocalDescriptions.properties
@@ -115,6 +115,7 @@ hawkular-wildfly-agent.managed-servers.remote-dmr.add=Adds a remote DMR managed 
 hawkular-wildfly-agent.managed-servers.remote-dmr.remove=Removes a remote DMR managed server
 hawkular-wildfly-agent.managed-servers.remote-dmr.name=A name this agent will assign to the remote application server
 hawkular-wildfly-agent.managed-servers.remote-dmr.enabled=True if you want to monitor the resources hosted in this server
+hawkular-wildfly-agent.managed-servers.remote-dmr.protocol=If specified, the protocol to use to connect where the remote application server management interface is listening. For example "remote". This ignores the value of use-ssl. If not specified, the default is "http-remoting" unless use-ssl is true in which case the default is "https-remoting".
 hawkular-wildfly-agent.managed-servers.remote-dmr.host=The host where the remote application server management interface is listening on
 hawkular-wildfly-agent.managed-servers.remote-dmr.port=The port where the remote application server management interface is listening to
 hawkular-wildfly-agent.managed-servers.remote-dmr.username=Name of the management user that can connect to the application server management interface

--- a/hawkular-wildfly-agent/src/main/resources/schema/hawkular-agent-monitor-subsystem.xsd
+++ b/hawkular-wildfly-agent/src/main/resources/schema/hawkular-agent-monitor-subsystem.xsd
@@ -232,6 +232,7 @@
   <xs:complexType name="remoteDmrType">
     <xs:attribute name="name"                  type="xs:string" use="required"/>
     <xs:attribute name="enabled"               type="xs:boolean"/>
+    <xs:attribute name="protocol"              type="xs:string"/>
     <xs:attribute name="host"                  type="xs:string" use="required"/>
     <xs:attribute name="port"                  type="xs:int"    use="required"/>
     <xs:attribute name="username"              type="xs:string"/>

